packages/babel-traverse/src/visitors.ts

```diff
@@ -245,8 +245,14 @@ export function merge(
   states: any[] = [],
   wrapper?: VisitWrapper | null,
 ): ExplodedVisitor {
-  // @ts-expect-error don't bother with internal flags so it can work with earlier @babel/core validations
-  const mergedVisitor: ExplodedVisitor = {};
+  const mergedVisitor: ExplodedVisitor = { _verified: true, _exploded: true };
+  if (!process.env.BABEL_8_BREAKING) {
+    // For compatibility with old Babel versions, we must hide _verified and _exploded.
+    // Otherwise, old versions of the validator will throw sayng that `true` is not
+    // a function, because it tries to validate it as a visitor.
+    Object.defineProperty(mergedVisitor, "_exploded", { enumerable: false });
+    Object.defineProperty(mergedVisitor, "_verified", { enumerable: false });
+  }
```

Comment:
I'm surprised they would fail for enumerable `_exploded` but still work for non-enumerable `_exploded`.

Reply:
I have some concerns that visitors will still be revalidated on these versions. (I'm not sure how older versions did it)
Since they throw on `_exploded`, that means they probably don't follow `_exploded` and validate again.

Reply:
They check `visitor._exploded` and `visitor._verified`, so it works even if it's non-enumerable.

Reply:
They check `visitor._verified`, so it works even if it's non-enumerable.